app/src/sharedTest/java/org/oppia/app/recyclerview/BindableAdapterTest.kt

```diff
@@ -251,6 +251,62 @@ class BindableAdapterTest {
     }
   }
 
+  @Test
+  fun testBindableAdapter_incomingSameData_noRebindingShouldHappen() {
```

Comment:
This test is definitely on the right track, but it only tests a single behavior (when setting the same data, no rebinding happens). It's also useful to test the other benefits that we get from the diff, such as:
- Inserting a new item triggers an insertion rather than rebinding the whole list
- Removing an item just changes that one item
- Moving an item has two simple operations instead of recreating the whole list
- Change the content of an item but not the item's identifier causes just that item to specify that it's changed

Can you add tests for each of these? Consider how to reuse the existing functionality in this test via helpers to keep the tests narrowly focused on illustrating the behavior being tested.

Reply:
Done, 
I added test cases which are passing and failing with and without DiffUtil implementation. 
Only thing I see here is, It is passing on Robolectric only, not espresso.